Extensions/CompositeMetadata.md

```diff
@@ -0,0 +1,45 @@
+# Composite Metadata Extension
+
+_This extension specification is currently incubating._
+
+## Introduction
+There are a number of situations where an arbitrary collection of discrete metadata types should be attached to frame.  For example, a request frame may want to include both routing metadata as well as tracing metadata.  This extension specification provides an interoperable structure for metadadata payloads to contain multiple discrete metadata types.  It is designed such that if a consumer of the metadata is unaware of a particular type, it can be safely skipped and the next one read.
+
+## Metadata Payload
+This metadata type **MUST** only be used in frame types used to initiate streams.  This includes [`REQUEST_FNF`][rf], [`REQUEST_RESPONSE`][rr], [`REQUEST_STREAM`][rs], and [`REQUEST_CHANNEL`][rc].  The [`SETUP` Frame][s] Metadata MIME Type is `message/x.rsocket.composite-metadata`.
```

Comment:
Nit: “initiate streams” doesn’t sit well with me for rf and rr.

Reply:
I'll change it to "...to initiate interactions."

As to not `SETUP` and not `PAYLOAD`, on the call that resulted in this PR we discussed the idea that the proper level of granularity for this kind of extension was per-stream.  You'd set tracing metadata, or routing metadata on a per-stream basis and it didn't make sense for either connection-level or individual payloads. The plan was to take these three specs into incubation with this restriction and start using them to see if that hypothesis held.  If it didn't we broaden the spec to include other granularity.